src/main/java/org/kitteh/irc/client/library/feature/sts/STSPropertiesStorageManager.java

```diff
@@ -0,0 +1,199 @@
+        } catch (IOException e) {
+            throw new KittehSTSException(e.getMessage());
+        }
+    }
+
+    /**
+     * Gets an STS policy from the store, looking it up via hostname.
+     *
+     * @param hostname the hostname (as sent in the SNI by the client)
+     * @return all data sent by the server in the CAP LS "sts" value when we connected securely
+     */
+    @Override
+    @Nonnull
+    public Optional<STSPolicy> getEntry(@Nonnull String hostname) {
+        Sanity.nullCheck(hostname, "A valid hostname must be provided for this entry.");
+
+        this.pruneEntries();
+        if (!this.hasEntry(hostname)) {
+            return Optional.empty();
+        }
+
+        String value = this.properties.getProperty(hostname);
+        String[] components = value.split("; ");
+        String data = components[1];
+        return Optional.of(STSUtil.getSTSPolicyFromString(",", data));
+    }
+
+    /**
+     * Checks if a policy has been stored for the hostname.
+     *
+     * @param hostname the hostname to check
+     * @return whether the entry exists in the store
+     */
+    @Override
+    public boolean hasEntry(@Nonnull String hostname) {
+        Sanity.nullCheck(hostname, "A valid hostname must be provided for this entry.");
+
+        this.pruneEntries();
+        return this.properties.containsKey(hostname);
+    }
+
+    /**
+     * Cleans up the outdated entries from the properties instance.
+     */
+    private void pruneEntries() {
+        Set<String> stagedRemovals = new HashSet<>();
+
+        for (String hostname : this.properties.stringPropertyNames()) {
+            String value = this.properties.getProperty(hostname);
+            String[] components = value.split("; ");
+            ZonedDateTime dt = ZonedDateTime.parse(components[0], DATE_TIME_FORMATTER);
+            if (dt.isBefore(ZonedDateTime.now(ZoneOffset.UTC))) {
+                stagedRemovals.add(hostname); // CME
+            }
+        }
+
+        stagedRemovals.forEach(this.properties::remove);
+        this.saveData();
+    }
+
+    /**
+     * Deletes an entry from the store (used for 0 duration policies).
+     * <p>
+     * Implementers MUST ignore requests to remove entries that do not exist.
+     *
+     * @param hostname the hostname to remove the policy for
+     */
+    @Override
+    public void removeEntry(@Nonnull String hostname) {
+        Sanity.nullCheck(hostname, "A valid hostname must be provided for this entry.");
+
+        if (!this.hasEntry(hostname)) {
+            return;
+        }
+        this.properties.remove(hostname);
+    }
+
+    /**
+     * Reserialize the STS policy in the same form as used in the spec.
+     *
+     * @param policy The map of keys -> optional string values
+     * @return a serialized string
+     */
+    private String reserializeData(STSPolicy policy) {
+        StringBuilder sb = new StringBuilder(policy.getOptions().size()*10+policy.getFlags().size()*5);
```

Comment:
It depends on how intuitively you grok the rules of operator precedence and how apparent it is that the multiplication happens before the addition, I guess.

For what it's worth, this is hardly a crucial calculation and simply a guess at the potential size of the string using a guess of `key=val` has an avg length of 10 characters and a flag has an avg length of 5 characters.

This is to avoid needlessly numerous resizes of the `StringBuilder`'s backing `char[]` store which will involve a new memory allocation and copy of all the characters.